File: docmanager/src/main/java/com/doceditor/docmanager/kafka/dto/SnapshotMessage.java
package com.doceditor.docmanager.kafka.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SnapshotMessage {
    private String type;
    private String documentId;
    private String documentString;
    private Long lamportClock;
    private Long timestamp;

}

File: connection-manager/messages/snapshot_message.go
package messages

type SnapshotMessage struct {
	Type           string `json:"type"`
	DocumentId     string `json:"documentId"`
	DocumentString string `json:"documentString"`
	LamportClock   int64  `json:"lamportClock"`
	Timestamp      int64  `json:"timestamp"`
}
